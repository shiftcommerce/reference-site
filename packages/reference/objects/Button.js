// Libraries
import { Component } from 'react'
import classNames from 'classnames'

class Button extends Component {
  render () {
    let {
      status,
      label,
      size,
      disabled,
      className,
      ...otherProps
    } = this.props

    disabled = disabled || false
    status = status || 'primary'
    size = size || 'lrg'

    return (
      <button
        className={classNames(className, 'o-button', `o-button--${status}`, `o-button--${size}`, {'o-button--disabled': disabled})}
        tabIndex='0'
        role='button'
        disabled={disabled}
        {...otherProps}
      >
        { label }
      </button>
    )
  }
}

export default Button
