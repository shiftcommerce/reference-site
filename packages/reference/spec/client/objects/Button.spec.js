// Object
import Button from '../../../client/objects/Button'

test('renders correctly', () => {
  // Arrange & Act
  const wrapper = mount(
    <Button label='ADD TO BAG' status='positive' />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
})

test('disabled button', () => {
  // Arrange & Act
  const wrapper = mount(
    <Button label='Testing Button' disabled='true' />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('button')).toHaveClassName('o-button--disabled')
  expect(wrapper.find('button')).toHaveProp('disabled', 'true')
})

describe('Button by default', () => {
  test('disabled should be false', () => {
    // Arrange & Act
    const wrapper = mount(
      <Button label='Enabled Button' />
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('button')).not.toHaveClassName('o-button--disabled')
  })

  test('should be of large size', () => {
    // Arrange & Act
    const wrapper = mount(
      <Button label='Large Button' />
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('button')).toHaveClassName('o-button--lrg')
  })

  test('status should be primary', () => {
    // Arrange & Act
    const wrapper = mount(
      <Button label='Primary Button' />
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('button')).toHaveClassName('o-button--primary')
  })
})
