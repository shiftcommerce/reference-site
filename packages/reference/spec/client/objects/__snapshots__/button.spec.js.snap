// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button by default disabled should be false 1`] = `
<Button
  label="Enabled Button"
  size="lrg"
  status="primary"
>
  <div
    className="o-button_container"
  >
    <button
      className="o-button o-button--primary o-button--lrg"
      disabled={false}
      role="button"
      tabIndex="0"
    >
      Enabled Button
    </button>
  </div>
</Button>
`;

exports[`Button with props should be of large size 1`] = `
<Button
  label="Large Button"
  size="lrg"
>
  <div
    className="o-button_container"
  >
    <button
      className="o-button o-button--lrg"
      disabled={false}
      role="button"
      tabIndex="0"
    >
      Large Button
    </button>
  </div>
</Button>
`;

exports[`Button with props status should be primary 1`] = `
<Button
  label="Primary Button"
  status="primary"
>
  <div
    className="o-button_container"
  >
    <button
      className="o-button o-button--primary"
      disabled={false}
      role="button"
      tabIndex="0"
    >
      Primary Button
    </button>
  </div>
</Button>
`;

exports[`disabled button 1`] = `
<Button
  disabled="true"
  label="Testing Button"
  size="lrg"
  status="primary"
>
  <div
    className="o-button_container"
  >
    <button
      className="o-button o-button--primary o-button--lrg o-button--disabled"
      disabled="true"
      role="button"
      tabIndex="0"
    >
      Testing Button
    </button>
  </div>
</Button>
`;

exports[`renders correctly 1`] = `
<Button
  label="ADD TO BAG"
  size="lrg"
  status="positive"
>
  <div
    className="o-button_container"
  >
    <button
      className="o-button o-button--positive o-button--lrg"
      disabled={false}
      role="button"
      tabIndex="0"
    >
      ADD TO BAG
    </button>
  </div>
</Button>
`;
